x-pack/test_serverless/shared/services/user_manager/svl_user_manager.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { REPO_ROOT } from '@kbn/repo-info';
+import { resolve } from 'path';
+import * as fs from 'fs';
+import Url from 'url';
+import { load as loadYaml } from 'js-yaml';
+import { Cookie } from 'tough-cookie';
+import { createNewSAMLSession, createSessionWithFakeSAMLAuth } from './saml_auth';
+import { FtrProviderContext } from '../../../functional/ftr_provider_context';
+
+export interface User {
+  readonly username: string;
+  readonly password: string;
+}
+
+export type Role = string;
+
+export class Session {
+  readonly cookie;
+  readonly username;
+  readonly fullname;
+  constructor(cookie: Cookie, username: string, fullname: string) {
+    this.cookie = cookie;
+    this.username = username;
+    this.fullname = fullname;
+  }
+
+  getCookieValue() {
+    return this.cookie.value;
+  }
+}
+
+export function SvlUserManagerProvider({ getService }: FtrProviderContext) {
+  const kibanaServer = getService('kibanaServer');
+  const config = getService('config');
+  const log = getService('log');
+  const isServerless = config.get('serverless');
+  const isCloud = !!process.env.TEST_CLOUD;
+  const cloudRoleUsersFilePath = resolve(REPO_ROOT, '.ftr', 'role_users.json');
```

Comment:
Can we move this out out of the repo or .gitignore?

Reply:
@pheyos what do you think is the best way to proceed?
1. I can completely hide it and read from env var, e.g. TEST_CLOUD_ROLE_USERS_PATH
2. add `role_users.json` to .gitignore so nobody accidentally exposes their cloud credentials

The 2nd feels more DevUX-friendly to me.

Reply:
@dmlemeshko I think adding it to .gitignore is good enough.

Reply:
fixed in 36f539716cd3bdc5d19550289f3bfdd4b33c1b35